ttnn/cpp/ttnn/operations/matmul/device/matmul_op_multi_core_reuse_mcast_1d_program_factory.cpp

```diff
@@ -1659,8 +1989,8 @@ operation::ProgramWithCallbacks matmul_multi_core_reuse_mcast_1d_optimized_(
 
     if (fp32_dest_acc_en) {
         TT_FATAL(
-            out_subblock_h * out_subblock_w <= 4,
-            "Total number of tiles in a subblock must be less than 4 when in fp32_dest_acc mode");
+            out_subblock_h * out_subblock_w <= 8,
```

Comment:
Why change this to 8?

Isn't 16 cut in half because of half dest mode and in half again because of fp32?

If anything this could be removed since 
```
                TT_FATAL(
                    (program_config.out_subblock_w * program_config.out_subblock_h) <= available_reg_count,
                    "out_subblock_w {} times out_subblock_h {} needs to be at most {} to fit in hardware",
                    program_config.out_subblock_w,
                    program_config.out_subblock_h,
                    available_reg_count);
```
has been added to validate() in matmul_op.cpp

Reply:
So one of the remaining todo's in the PR is getting support for full dest mode (see [here](https://github.com/tenstorrent/tt-metal/pull/14964#:~:text=Get%20official%20support%20for%20full%20dest%20when%20fp32_accum_mode%20%3D%20True%20(this%20is%20not%20necessary%20but%20leads%20to%20significant%20perf%20gains))). If it isn't merged before this PR, I can revert the hard coded value.

In my testing for the shapes used in the Llama models, I have found that we can actually use 8 here, and it results in significant speedups since there's no reload.

Reply:
seems like full dest mode support has already been added in main. However, the pybind for the compute kernel config has not been updated to allow users to enable it.

I've opened a PR [here](https://github.com/tenstorrent/tt-metal/pull/15007) that updates the pybind, and I am currently waiting for confirmation from @amahmudTT that this change is correct.